pallets/qa-tools/src/pallet_tools/order_book.rs

```diff
@@ -1,26 +1,72 @@
 use crate::Config;
-use common::{balance, AssetInfoProvider, Balance, PriceVariant};
+use common::{AssetInfoProvider, Balance, PriceVariant};
 use frame_support::pallet_prelude::*;
 use frame_support::sp_runtime::traits::Zero;
 use frame_support::traits::Time;
 use frame_system::pallet_prelude::*;
 use order_book::DataLayer;
 use order_book::{MomentOf, OrderBook, OrderBookId};
 use order_book::{OrderPrice, OrderVolume};
+use rand::{Rng, SeedableRng};
+use rand_chacha::ChaCha8Rng;
+use sp_std::iter::repeat;
 use sp_std::prelude::*;
 
-#[derive(Encode, Decode, Clone, PartialEq, Eq, scale_info::TypeInfo)]
-#[cfg_attr(feature = "std", derive(Debug))]
-pub struct OrderBookFillSettings<Moment> {
-    /// Best (highest) price for placed buy orders
-    pub best_bid_price: order_book::types::OrderPrice,
-    /// Best (lowest) price for placed sell orders
-    pub best_ask_price: order_book::types::OrderPrice,
-    /// Lifespan of inserted orders, max by default
-    pub lifespan: Option<Moment>,
+pub mod settings {
+    use codec::{Decode, Encode};
+    use common::Balance;
+    use order_book::OrderVolume;
+
+    /// Parameters for filling one order book side
+    #[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, scale_info::TypeInfo)]
+    pub struct SideFill {
+        /// highest price for bids; lowest for asks
+        pub best_price: Balance,
+        /// lowest price for bids; highest for asks
+        pub worst_price: Balance,
+        pub price_step: Balance,
+        pub orders_per_price: u32,
+        /// Default: `min_lot_size..=max_lot_size`
+        pub amount_range_inclusive: Option<(Balance, Balance)>,
+    }
+
+    /// Parameters for orders amount generation
+    #[derive(Encode, Decode, Clone, PartialEq, Eq, scale_info::TypeInfo)]
+    #[cfg_attr(feature = "std", derive(Debug))]
+    pub struct RandomAmount {
+        min: OrderVolume,
+        max: OrderVolume,
+    }
+
+    impl RandomAmount {
+        pub fn new(min: OrderVolume, max: OrderVolume) -> Option<Self> {
+            if max >= min {
+                Some(Self { max, min })
+            } else {
+                None
+            }
+        }
+    }
+
+    #[derive(Encode, Decode, Clone, PartialEq, Eq, scale_info::TypeInfo)]
+    #[cfg_attr(feature = "std", derive(Debug))]
+    pub struct OrderBookFill<Moment, BlockNumber> {
+        /// Best price = highest, worst = lowest.
+        pub bids: SideFill,
+        /// Best price = lowest, worst = highest.
+        pub asks: SideFill,
+        /// Lifespan of inserted orders, max by default.
+        pub lifespan: Option<Moment>,
+        /// Seed for producing random values during the fill process. If `None`,
+        /// current block is chosen
+        pub random_seed: Option<BlockNumber>,
+    }
```

Comment:
@ra9mls can you check preliminary that such settings are ok? (they are to be set for each order book id in the extrinsic)